_posts/en/newsletters/2021-03-17-newsletter.md

```diff
@@ -0,0 +1,127 @@
+    cl2672] and LND in [Newsletter #92][news92 lnd4079]), developers may
+    want to give these sort of failure scenarios more attention.
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+- [HWI 2.0.0RC2][hwi 2.0.0] is a release candidate for the next major
+  version of HWI.  FIXME:harding
+
+- [Rust-Lightning 0.0.13][] FIXME:harding
+
+- [BTCPay Server 1.0.7.0][] FIXME:harding
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1
+repo], [Hardware Wallet Interface (HWI)][hwi repo],
+[Rust Bitcoin][rust bitcoin repo], [BTCPay Server][btcpay server repo],
+[Bitcoin Improvement Proposals (BIPs)][bips repo], and [Lightning
+BOLTs][bolts repo].*
+
+- [Bitcoin Core #21007][] bitcoind: Add -daemonwait option to wait for initialization FIXME: jnewbery
+
+- [C-Lightning #4404][] allows the `keysend` RPC (see [Newsletter
+  #107][news107 keysend]) to send messages even to nodes that don't
+  explicitly signal that they support the feature.  As
+  [discussed][fromknecht keysend], the signal was never standardized and
+  the procedure implemented by LND didn't depend on signaling, so this
+  change should allow C-Lightning to send to roughly the same set of
+  nodes tha LND can address.
+
+- [C-Lightning #4410][] brings the experimental implementation for
+  dual-funded channels into line with the most recent [draft
+  specification][bolts #851] changes.  Most notably, the use of Proof of
+  Discrete Log Equivalency (PODLE) has been dropped, at least
+  temporarily (see [Newsletter #83][news83 podle] for original
+  discussion of PODLEs and [Newsletter #131][news131 podle] for
+  discussion about alternatives).  Subsequent to this merge, a [new
+  PR][c-lightning #4427] was opened that will make experimentation with
+  dual-funding more accessible by eliminating the need to compile
+  C-Lightning with special build flags (although a special configuration
+  option will still be required).
+
+- [LND #5083][] allows a [PSBT][topic psbt] to be read from a file
+  rather than by reading the standard input (stdin) file descriptor.
+  Some terminals have a [limit][lnd #5080] on the number of characters
+  that can be added to stdin simultaneously (i.e. pasted), which made
+  PSBTs over 4096 characters (3 KiB) <!-- PSBTs are base64 encoded,
+  base64 provides 6 bits per character, so 4096 * 6 / 8 = 3072 = 3 KiB
+  --> unusable.  Especially now that several hardware wallets require
+  PSBTs include previous transactions for segwit spends (see [Newsletter
+  #101][news101 segwit overpayment]), it's easy to create PSBTs over 3
+  KiB in size.
```

Comment:
Thanks for the comment, I was indeed confused by characters that were smaller than a byte. :+1: 

However, I'm confused by the reference of both figures. Is the relevant limit the character count or the data amount?

If I recall correctly, transaction sizes in Bitcoin Core are explicitly tracked in kB (1000 B) rather than in KiB. So, if the figure was meant to relate to transaction sizes, it may make sense to rather express it in kB.

Reply:
1. The relevant limit is the character count.  Pastes of anything into the terminal (including PSBTs) will be truncated to 4096 characters.  E.g., if you paste hex a hex string, it'll be truncated to 4096 characters == 2048 bytes.  I'll try to make this clearer in the text.

2. BIP174 PSBTs don't have a direct relationship to actual transaction sizes since PSBTs also include data like entire previous transactions for each input in order to calculate fees, so I think their size can be expressed however we want.  In this case, I used kibibytes because it allowed me be precise instead of saying "about 3 kB".  I don't have a strong preference for units here, so let me know if you'd like it changed.

Reply:
maybe "3072 bits" as KiB might be a bit obscure for laypeople (personally, I've always thought of KB in binary terms, technically incorrect as it may be)